pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -362,8 +362,14 @@ def _get_tools(
                         raise UserError(
                             "`ImageGenerationTool` is not supported by this model. Use a model with 'image' in the name instead."
                         )
-                    if tool.aspect_ratio:
-                        image_config = ImageConfigDict(aspect_ratio=tool.aspect_ratio)
+                    # Build image_config from tool parameters, only including non-None values
+                    image_config_kwargs: dict[str, Any] = {}
+                    if tool.aspect_ratio is not None:
+                        image_config_kwargs['aspect_ratio'] = tool.aspect_ratio
+                    if tool.resolution is not None:
+                        image_config_kwargs['image_size'] = tool.resolution
+                    if image_config_kwargs:
+                        image_config = ImageConfigDict(**image_config_kwargs)
```

Comment:
Since `image_config` is already typed as `ImageConfigDict`, which is a `TypedDict`, we can drop `image_config_kwargs`, set `image_config` to `{}`, and then set dict fields on it as usual.